### PR TITLE
Reorganization - sanitzing lib folder

### PR DESCRIPTION
**Pull request by fearphage**
This does a lot. I would normally submit multiple PRs for this, but I thought I'd try to one-shot it. This does several things:
- move sample files from `lib` to `examples` folder
- replaced `gulp-eslint` with `eslint`
- moved eslint config to `package.json` (and test eslint config to `test/.eslintrc.json`)
- all files in `lib` pass eslint now
- using `mocha.opts` file for mocha options
  
  Let me know what you think.

This is related to comments in (and supercedes I believe) #314.

### CODE REVIEW

**Comment by albertjan** [comment]
priorityQueue is not an example is it?

**Comment by fearphage** [comment]
You should add a `.travis.yml` file. It would help contributors know what versions to test against.

**Comment by fearphage** [comment]
> priorityQueue is not an example is it?

It wasn't referenced in `lib/queue.js` so I thought it was. I honestly based that on `cluster-queue.js` and `state.js`.

**Comment by fearphage** [comment]
And it also was not being `lint`ed.

**Comment by fearphage** [comment]
@albertjan After looking more closely, you are correct. That would explain the test also. :wink:

**Comment by manast** [comment]
sorry for super late reply. I like many things in this PR. Can be merged for the next version after I fix the conflicts.

File: package.json [hunk 1]
```diff
@@ -27,14 +27,38 @@
...
   "scripts": {
-    "test": "gulp && mocha test/test_* --reporter spec --timeout 5000",
+    "lint": "eslint lib test",
+    "pretest": "npm run lint",
+    "test": "mocha",
     "postpublish": "git push && git push --tags"
+  },
+  "eslintConfig": {
+    "rules": {
+      "indent": [
```
Comment (bradvogel) [inline package.json]: Can you add: `'semi': 2`

File: package.json [hunk 2]
```diff
@@ -27,14 +27,38 @@
...
+          "SwitchCase": 1
+        }
+      ],
+      "valid-jsdoc": 0,
+      "func-style": 0,
+      "no-use-before-define": 0,
+      "camelcase": 1,
+      "no-unused-vars": 1,
+      "no-alert": 1,
+      "no-console": 1,
```
Comment (bradvogel) [inline package.json]: Can you make this `[2, { allow: ["warn", "error"] }]`?

**Comment by bradvogel** [comment]
@fearphage Any chance you can rebase this on the `v3.0.0` branch and retarget the PR to merge into that branch? Branch `v3.0.0` is where all active development is currently, and will soon be merged back into `master`. That would really help us!

File: package.json [hunk 3]
```diff
@@ -27,14 +27,38 @@
...
+    "eslint": "^2.13.1",
     "expect.js": "^0.3.1",
-    "gulp": "^3.9.1",
-    "gulp-eslint": "^2.1.0",
     "mocha": "^2.5.3",
     "sinon": "^1.17.7"
   },
   "scripts": {
-    "test": "gulp && mocha test/test_* --reporter spec --timeout 5000",
+    "lint": "eslint lib test",
```
Comment (bradvogel) [inline package.json]: Can you have it ignore `lib/priority-queue.js`, as is done here: https://github.com/OptimalBits/bull/pull/515/files#diff-b9e12334e9eafd8341a6107dd98510c9R12

File: package.json (hunk 2 above)
Comment (bradvogel) [inline package.json]: Comment out `console.logs` used in tests, if any.

File: package.json (hunk 1 above)
Comment (fearphage) [inline package.json]: Done.

File: package.json (hunk 2 above)
Comment (fearphage) [inline package.json]: Done.

File: package.json (hunk 3 above)
Comment (fearphage) [inline package.json]: Does it need to be ignored? It's passing now.

**Comment by fearphage** [comment]
I pushed all the changes. Feel free to rebase. It looks messy.

**Comment by bradvogel** [comment]
Thanks! Can you switch the target branch to `v3.0.0` and rebase on that branch?

**Comment by bradvogel** [comment]
Oh I see your comment - yeah, if you don't have time to clean the new versions of those files (in v3.0.0), I can. No problem

File: package.json (hunk 3 above)
Comment (manast) [inline package.json]: It will be removed very very soon.

**Comment by bradvogel** [comment]
Ok, filed new PR https://github.com/OptimalBits/bull/pull/516 that applies this to the v3.0.0 branch. This PR can be closed.

**Pull request closed without merging**